docs/Authenticating-Users-via-Tokens.md

```diff
@@ -0,0 +1,89 @@
+| USERNAME   | varchar(255) | NO   | MUL | NULL    |       |
+| EXPIRATION | datetime     | NO   |     | NULL    |       |
+| CREATION   | datetime     | YES  |     | NULL    |       |
++------------+--------------+------+-----+---------+-------+
+4 rows in set (0.00 sec)
+```
+In addition to storing the token (the unique number), its creation and expiration dates along with the associated username is also stored prior to presenting it to the user.  When the data access token is presented in a subsequent web service request, the cBioPortal compares the token presented in the request against this table for validity.  If the token exists in the table and has not yet expired, the request will be satisfied.  The creation date is used by the cBioPortal when a user requests a new data access token.  Based on the server configuration, if the number of outstanding data access tokens together with this new one exceeds the number of allowable data access tokens per user, the cBioPortal may refuse to present a new one or it may invalidate the oldest one before presenting the new one to the user.
+
+### Modifying Configuration
+The following properties must be present in portal.properties in order to allow direct access to the cBioPortal web service when login is required:
+
+**Property**: dat.method (required)
+- **Description**: The desired data access token implementation.
+- **Permissible Values**: jwt, uuid, none
+- **Default Value**: none
+
+**Property**: dat.unauth_users (optional)
+- **Description**: A list of users that should not be allowed to download a data access token.
+- **Permissible Values**: A comma-delimited list of valid user names as found in the users table.
+- **Default Value**: anonymousUser
+
+**Property**: dat.ttl_seconds (required)
+- **Description**: The time in seconds between token creation and token expiration.
+- **Permissible Values**: An integer value greater than zero.
+- **Default Value**: -1
+
+**Property**: dat.jwt.secret_key (required only when dat.method = jwt)
+- **Description**: The signature used to sign JWT based data access tokens during creation.  This signature is used in subsequent web service requests to validate the request.  This property is used only when dat.method = jwt.
+- **Permissible Values**: A string value.
+- **Default Value**: none
+
+**Property**: dat.uuid.max_number_per_user (required only when dat.method = uuid)
+- **Description**: This property determines the number of data access tokens that can be allocated to a single user at any given time.  If a user requests a data access token that puts them over this limit, then based on the value of dat.uuid.revoke_other_tokens (see below), the cBioPortal will either refuse to present a new token to the user or it may invalidate the oldest one before presenting the new one to the user.  This property is used only when dat.method = uuid.
+- **Permissible Values**: An integer value greater than zero.
+- **Default Value**: -1
```

Comment:
What happens with a default value of -1...unlimited number of tokens?  I think we should have the default value to 1 and set revoke_other_tokens default to true.